clang/lib/Lex/Lexer.cpp

```diff
@@ -4589,6 +4589,9 @@ bool Lexer::LexDependencyDirectiveToken(Token &Result) {
 
   if (Result.is(tok::hash) && Result.isAtStartOfLine()) {
     PP->HandleDirective(Result);
+    if (PP->hadModuleLoaderFatalFailure())
+      // With a fatal failure in the module loader, we abort parsing.
+      return true;
```

Comment:
Here `true` means "stop lexing" and it is interpreted in the loop
```c++
  while (!CurLexerCallback(*this, Result))
    ;
```

in `Preprocessor::Lex`.